Code Change: changed types of configuration struct (refer to issue #34)

minovations opened the pull request:
Code Change: changed types of configuration struct (refer to issue #34)

Changes:
- libconfiguration/Makefile.am
  - Added include dir libcommon
- libconfiguration/include/libconfiguration.h
  - Changed types of configuration pointers
- source/gui/mainwindow.c
  - renamed the macro's and names of the pointers

Commit:
Code Change: changed types of configuration struct (refer to issue #34)

Changes:
* libconfiguration/Makefile.am
  - Added include dir libcommon

* libconfiguration/include/libconfiguration.h
  - Changed types of configuration pointers

* source/gui/mainwindow.c
  - renamed the macro's and names of the pointers

roelj commented:
Why are we now keeping track of Tree pointers instead of Series and Studies directly? Do we need a Tree pointer to do Tree manipulation for active layers?

Commit:
Store program state in libconfiguration.

Changes made in files related to create a reference map file:
* libconfiguration/Makefile.am
  - Changed include directories of correct types
* libconfiguration/include/libconfiguration.h
  - Removed not related program state features.
  - Added serietree pointer.
  - replaced other pointers to pixeldata and mainwindowgui
* libpixeldata/Makefile.am
  - Removed configuration
* libpixeldata/include/libpixeldata.h
  - Added correct documentation of new function
* libpixeldata/src/libpixeldata.c
  - Added get function for returning pointer of list
* source/gui/mainwindow.c
  - changed macronames as specified in libconfiguration.

Commit:
Tool to create a graphic plot of the internal dependencies between modules

* Makefile.am
  - added functionality
* configure.ac
  - added some options
* dependencies.sh
  - a bash script to convert the generated map file to a dot file. The dot file could be converted to a graphic with graphviz, a free usable tool.

```bash
$ make clean
$ autoreconf -i
$ ./configure --enable-cref-map=yes
$ make
$ ./dependencies.sh clmedview.map clmedview.dot
$ dot -Tsvg clmedview.dot -o clmedview.svg
```
